msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -499,6 +532,25 @@ public List getAllUserStatistics(){
 		//clear any gradebook info:
 		resetGradebookVariables();
 
+		// Determine whether the site is a pure anonymous scenario (Ie. all topics in the site are anonymous)
+		String siteId = ToolManager.getCurrentPlacement().getContext();
+		List<Topic> allTopics = forumManager.getTopicsInSite(siteId);
```

Comment:
Yeah, I think it would deliver performance improvements in the common case to prevent the propagation of all the topics in the site from the DB, but it's probably not critical. I'll create a ticket for our OWL roadmap.